File: backend/services/aktivnostiService.js
const db = require('../models');

class AktivnostiService {
  async getAllAktivnosti() {
    try {
      return await db.aktivnost.findAll({
        where: { isActive: true },
        order: [['id', 'ASC']]
      });
    } catch (error) {
      throw new Error(`Error fetching aktivnosti: ${error.message}`);
    }
  }

  async getAktivnostById(id) {
    try {
      const aktivnost = await db.aktivnost.findByPk(id);

      if (!aktivnost || !aktivnost.isActive) {
        throw new Error('Aktivnost not found');
      }

      return aktivnost;
    } catch (error) {
      throw new Error(`Error fetching aktivnost: ${error.message}`);
    }
  }

  async createAktivnost(data) {
    try {
      return await db.aktivnost.create({
        aktivnost: data.aktivnost,
        rokTrajanja: data.rokTrajanja,
        cijena: data.cijena !== undefined ? data.cijena : null,
        isActive: true
      });
    } catch (error) {
      throw new Error(`Error creating aktivnost: ${error.message}`);
    }
  }

  async updateAktivnost(id, data) {
    try {
      const aktivnost = await db.aktivnost.findByPk(id);

      if (!aktivnost || !aktivnost.isActive) {
        throw new Error('Aktivnost not found');
      }

      return await aktivnost.update({
        aktivnost: data.aktivnost !== undefined ? data.aktivnost : aktivnost.aktivnost,
        rokTrajanja: data.rokTrajanja !== undefined ? data.rokTrajanja : aktivnost.rokTrajanja,
        cijena: data.cijena !== undefined ? data.cijena : aktivnost.cijena
      });
    } catch (error) {
      throw new Error(`Error updating aktivnost: ${error.message}`);
    }
  }

  async deleteAktivnost(id) {
    try {
      const aktivnost = await db.aktivnost.findByPk(id);

      if (!aktivnost || !aktivnost.isActive) {
        throw new Error('Aktivnost not found');
      }

      return await aktivnost.update({ isActive: false });
    } catch (error) {
      throw new Error(`Error deleting aktivnost: ${error.message}`);
    }
  }
}

module.exports = new AktivnostiService();

File: JavaSimpleBackend/src/main/java/com/atesti/service/AktivnostiService.java
package com.atesti.service;

import com.atesti.dto.AktivnostRequest;
import com.atesti.entity.Aktivnost;
import com.atesti.exception.BadRequestException;
import com.atesti.exception.ResourceNotFoundException;
import com.atesti.repository.AktivnostRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class AktivnostiService {

    private final AktivnostRepository aktivnostRepository;

    public List<Aktivnost> getAllAktivnosti() {
        return aktivnostRepository.findByIsActiveTrueOrderByIdAsc();
    }

    public Aktivnost getAktivnostById(Long id) {
        Aktivnost aktivnost = aktivnostRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Aktivnost not found"));

        if (!aktivnost.getIsActive()) {
            throw new ResourceNotFoundException("Aktivnost not found");
        }

        return aktivnost;
    }

    public Aktivnost createAktivnost(AktivnostRequest request) {
        if (request.getAktivnost() == null || request.getRokTrajanja() == null) {
            throw new BadRequestException("Error creating aktivnost: aktivnost and rokTrajanja are required");
        }

        Aktivnost aktivnost = Aktivnost.builder()
                .aktivnost(request.getAktivnost())
                .rokTrajanja(request.getRokTrajanja())
                .cijena(request.getCijena())
                .isActive(true)
                .build();

        return aktivnostRepository.save(aktivnost);
    }

    public Aktivnost updateAktivnost(Long id, AktivnostRequest request) {
        Aktivnost aktivnost = aktivnostRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Aktivnost not found"));

        if (!aktivnost.getIsActive()) {
            throw new ResourceNotFoundException("Aktivnost not found");
        }

        if (request.getAktivnost() != null) {
            aktivnost.setAktivnost(request.getAktivnost());
        }
        if (request.getRokTrajanja() != null) {
            aktivnost.setRokTrajanja(request.getRokTrajanja());
        }
        if (request.getCijena() != null) {
            aktivnost.setCijena(request.getCijena());
        }

        return aktivnostRepository.save(aktivnost);
    }

    public Aktivnost deleteAktivnost(Long id) {
        Aktivnost aktivnost = aktivnostRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Aktivnost not found"));

        if (!aktivnost.getIsActive()) {
            throw new ResourceNotFoundException("Aktivnost not found");
        }

        aktivnost.setIsActive(false);
        return aktivnostRepository.save(aktivnost);
    }
}
